src/app.rs

```diff
@@ -1608,20 +1611,21 @@ impl TaskwarriorTui {
     }
   }
 
-  fn get_task_files_max_mtime(&self) -> Result<SystemTime> {
-    let data_dir = shellexpand::tilde(&self.config.data_location).into_owned();
-    ["backlog.data", "completed.data", "pending.data"]
-      .iter()
-      .map(|n| fs::metadata(Path::new(&data_dir).join(n)).map(|m| m.modified()))
-      .filter_map(Result::ok)
-      .filter_map(Result::ok)
-      .max()
-      .ok_or_else(|| anyhow!("Unable to get task files max time"))
+  fn get_task_database_mtime(&self) -> Result<SystemTime> {
+    let data_dir = shellexpand::tilde(&self.config.data_location);
+    let database_path = Path::new(data_dir.as_ref()).join("taskchampion.sqlite3");
```

Comment:
I haven't had the chance to look into taskwarrior v3 yet, but isn't taskchampion an entirely different tool?

Reply:
The project is a bit hidden in the subfolder of the taskwarrior repository ([taskchampion README](https://github.com/GothenburgBitFactory/taskwarrior/tree/develop/taskchampion#taskchampion)).
As far as I understood it used to be it's own separate tool up until 2 years ago.

Taskchampion is now the new storage backend for taskwarrior, as far as I understand it.

To quote: 
> TaskChampion implements the task storage and synchronization behind Taskwarrior. It includes an implementation with Rust and C APIs, allowing any application to maintain and manipulate its own replica. It also includes a specification for tasks and how they are synchronized, inviting alternative implementations of replicas or task servers.
>
>See the [documentation](https://gothenburgbitfactory.github.io/taskwarrior/taskchampion/) for more!
>
>NOTE: Taskwarrior is currently in the midst of a change to use TaskChampion as its storage. Until that is complete, the information here may be out-of-date.

The taskchampion.sqlite3 [gets created](https://taskwarrior.org/docs/upgrade-3/) during the v2->v3 import phase, deprecating the .data files.

(The docs seem still a bit unstable, but I'm planning to read a bit more into taskchampions and libshared over the next weeks)

Reply:
Update: I ran into a problem with the tests, where a specific one fails after the taskwarrior update. I haven't fully understood the reason why yet, as I haven't looked through the testdata(and therefore don't quite get why test_draw_task_report_with_extended_modify_command expects there to be 26 tasks).
If I could ask you a few questions about the codebase after you get back from your vaccation(I don't know how to reach you outside of Github and asking every question on github gets a bit awkward), that would be great.

What will also (soft) block the merge for now is the changes which read-only commands now modify the database mtime(due to the new urgency setting getting updated). 
Softblocked because this sends two active tui instances in a loop of running update, therefore modifying the database which then triggers the update of the other instance.
(Is more than 1 instance active a supported use case, or did that run into bugs even before the upgrade?)

I really have to read through the taskchampion docu, the replika structure should be what is in-built to avoid such scenarios.

Reply:
The reason for the failing tests was most likely that I forgot to bump the taskwarrior version for CI/CD.
(As they ran error free locally)

Should be fixed now, it will require a manual approval from you to run though.

Reply:
> Thanks so much for making the PR and for working on this! I'm back from my vacation, happy to answer any questions.
> 
> In the meantime, I'll look into this more as well. I will update my local install to the latest version of taskwarrior to test this PR.

Glad that I can be of help!

What's the best way to reach you, this PR is getting a bit hard to read.
I could offer Matrix/Telegram/Signal/E-Mail(including GPG) etc.
We could alternatively use the Github Discussions feature.

(A first question would be the reasoning behind the current release options. Are incremental, debug and disabled lto a leftover from debugging?)

Reply:
Email is probably easiest! GitHub Discussions works too. I'm on the Ratatui discord if you want to ping me there that works too. I don't actively check Matrix but could sign up to if you want to chat there.

Reply:
I just sent you a DM in Discord(this message is meant as a second factor that I sent it. I'd like to avoid linking my Discord to my Github directly)